### PR TITLE
Spec clean up

### PR DESCRIPTION
**Pull request by cholmes**
**Related Issue(s):** #1012 #973 #822

**Proposed Changes:**

1. Consistency in how we refer to Item, Collection and Catalog
2. table of contents added
3. just link to spec constructs once, at the beginning. Link to the actual spec, not the readme.

**PR Checklist:**

- [x] This PR is made against the dev branch (all proposed changes except releases should be against dev, not master).
- [x] This PR has **no** breaking changes.
- [x] I have added my changes to the [CHANGELOG](https://github.com/radiantearth/stac-spec/blob/dev/CHANGELOG.md) **or** a CHANGELOG entry is not required.
- [x] This PR affects the [STAC API spec](https://github.com/radiantearth/stac-api-spec), and I have opened issue https://github.com/radiantearth/stac-api-spec/issues/105 to track the change.

### CODE REVIEW

**Comment by cholmes** [comment]
Ok, this is ready for review. I'm sure it could use another set of eyes on similar things. And I didn't go through everything for language - was mostly just updating capitalization and links, but tweaked things where I saw opportunities. But I'd say after review of these changes we should just merge this in and then someone else can take another pass in a new PR.

File: extensions/datacube/README.md
Comment (m-mohr) [inline extensions/datacube/README.md:91]:
```suggestion
- [Sentinel-2 Collection on openEO Google Earth Engine driver](https://earthengine.openeo.org/v1.0/collections/COPERNICUS/S2) (STAC 0.9.0)
```

File: extensions/datacube/README.md
Comment (m-mohr) [inline extensions/datacube/README.md:92]:
```suggestion
- [AGERA5 Collection on openEO VITO](https://openeo.vito.be/openeo/1.0/collections/AGERA5) (STAC 0.9.0)
```

File: extensions/sat/README.md
Comment (m-mohr) [inline extensions/sat/README.md:12]:
```suggestion
extensions that describe the actual data, such as the `eo` or `sar` extensions.
```

os is not a thing?!

**Review by m-mohr — APPROVED** [review]
Added three commitable suggestions, otherwise lgtm.

File: README.md
Comment (philvarner) [inline README.md:54]:
```suggestion
that enable clients to search for Item objects that match their filtering criteria. The four specifications are meant to be used 
```

Following my general belief that types should be "T objects" rather than "Ts" in most cases

File: README.md
Comment (philvarner) [inline README.md:67]:
```suggestion
In the context of STAC it is most likely a related group of STAC Items that is made available by a data provider.
```

set could be interpreted as to imply uniqueness (which is true, but not relevant here)

File: README.md
Comment (philvarner) [inline README.md:69]:
```suggestion
It enables discovery at a higher level than individual Item objects, providing a simple way to describe sets of data.
```

File: best-practices.md
Comment (philvarner) [inline best-practices.md:47]:
```suggestion
those who are creating STAC objects representing their data or creating tools to work with STAC. 
```

File: best-practices.md
Comment (philvarner) [inline best-practices.md:60]:
```suggestion
[Item](item-spec/item-spec.md) objects from the same server the page is on. [Cross-origin resource sharing](https://en.wikipedia.org/wiki/Cross-origin_resource_sharing),
```

File: extensions/sat/README.md
Comment (cholmes) [reply extensions/sat/README.md:12]: Lol. No idea...

**Review by jisantuc — APPROVED** [review]

**Pull request merged**
